docs: add `graphql-java-federation` to server implementation list

rkudryashov opened the pull request:
Hi,

This library is intended to be used with GraphQL Java services: https://github.com/rkudryashov/graphql-java-federation

trevor-scheer commented:
Hey @rkudryashov, this is interesting. Do you mind me asking what your reason is for forking `federation-jvm` instead of trying to have your changes upstreamed to the repo? It's fair to say our package hasn't received the attention it deserves, but @sachindshinde is working to change that and we'd certainly appreciate contributions to the project!

rkudryashov commented:
@trevor-scheer indeed it would be nice to merge this stuff into the main project, but I thought it will take too much time, because particularly I have reworked API to make it more convenient, upgraded major GraphQL Java version, made global refactoring, switched build tool from Maven to Gradle, etc. If my assumption was wrong, I could try to make PR to main repo.

rkudryashov commented:
Hi @trevor-scheer, please consider is it possible to include this library to the list. There was discussions after which it becomes clear that this library needs to be separated from its ancestor: https://github.com/apollographql/federation-jvm/pull/47, https://github.com/apollographql/federation-jvm/pull/48

Also, there is a working example on how to use it: https://github.com/rkudryashov/graphql-federation
